src/charm.py

```diff
@@ -353,7 +340,7 @@ def _prometheus_config(self):
 
         logger.debug("Prometheus config : {}".format(scrape_config))
 
-        return yaml.dump(scrape_config)
+        return yaml.dump(scrape_config, default_flow_style=True)
```

Comment:
I am surprised. I presume you really want to enforce "non block" style. I would have assumed this was the default.

Reply:
It's really because:

```
yaml.dump({
  "key": "[__value__]"
  })
```

Will spit out:
```
key: '[__value__]'
```

Which Prometheus won't parse. To match exactly as:
```
source_labels: [__address__]
```
Or:
```
source_labels: [juju_model, juju_model_uuid, ...]
```

"non block" style was required, since pyyaml considers '[' dangerous in YAML. After a little more testing once I knew it worked, I removed the flow. Prometheus parses block-style YAML arrays just fine, despite it not being in any docs. The last patch changed this back.